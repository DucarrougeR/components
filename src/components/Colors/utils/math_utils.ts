/**
 * Utility to convert a numerical value to a percent given the value and a max value.
 */
export const toPerc = (value: number, maxValue: number) =>
  Math.round((value / maxValue) * 100)
