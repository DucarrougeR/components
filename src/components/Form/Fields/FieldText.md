```js noeditor
const StatusAndResources = require('../../../../styleguide_components/StatusAndResources').StatusAndResources;

<StatusAndResources
  status="experimental"
  figmaURL="https://www.figma.com/file/2MG6DoSjk4IaLnjjFCnKFf/Forms"
  githubURL="https://github.com/looker/relens/blob/master/src/components/Form/Fields/FieldText.tsx"
  feedbackTitle="FieldText Component Feedback" />
```

```js
  <FieldText id="theme-name" name="themeName" label="Theme Name" />
```

```js

<form>
  <FieldText id="first_name" name="first_name" label="First Name" />
  <FieldText id="last_name" name="last_name" label="Last Name" />
  <FieldCheckbox id="subscribe" name="subscribe" label="Subscribe" />
  <Button>Submit</Button>
</form>
```
