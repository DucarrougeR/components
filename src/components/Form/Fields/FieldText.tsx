import * as React from 'react'
import { FieldJSX, FieldProps } from './Field'

export const FieldText: React.SFC<FieldProps> = ({
  id,
  label,
  name,
  alignLabel,
  validationState,
  alignValidationMessage,
  ...props
}) => {
  return FieldJSX<typeof props>(
    'FieldText',
    id,
    label,
    name,
    alignLabel,
    validationState,
    alignValidationMessage,
    props
  )
}
