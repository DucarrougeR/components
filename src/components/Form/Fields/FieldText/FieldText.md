```js noeditor
const StatusAndResources = require('../../../../../styleguide_components/StatusAndResources').StatusAndResources;

<StatusAndResources
  status="experimental"
  figmaURL="https://www.figma.com/file/2MG6DoSjk4IaLnjjFCnKFf/Forms"
  githubURL="https://github.com/looker/lens/blob/master/src/components/Form/Fields/FieldText.tsx"
  feedbackTitle="FieldText Component Feedback" />
```

### FieldText

The `<FieldText />` component is composed of an `<InputText />` component and a `<Label />` component. Using `<FieldText />` allows for rendering validation messages. By default, the label will render directly above the input field, which is the recommended convention, however this is adjustable with the `alignLabel` property.

```js
<FieldText name="firstName" label="First Name" />
<FieldText name="lastName" label=" Last Name" alignLabel="left" />
<FieldText name="requiredField" label="A required field" required />
```

### FieldText Validation

The `<FieldText />` component accepts `validationMessage` as a property. This is an object with properties `type` and `message`.

See `ValidationMessage` [here](/#!/ValidationMessage)

If you have multiple fields in need of validation, it is recommended that you wrap them in a `<Form />` component and pass to it a dictionary with key as field name and value as a `ValidationMessage`.

___Note: In the current implementation, any `ValidationMessage` set in a `<FieldText />` component will be overridden by `ValidationMessages` set in the parent `<Form />` component. Because of this, it is highly recommended that you only ever directly set `ValidationMessage` if a `<FieldText />` is outside a `<Form />`.___

```js
<FieldText name="someField" label="Some Field" validationMessage={{type: "error", message: "This is an error"}}/>
<FieldText name="someField2" label="Some Field2" validationMessage={{type: "warning", message: "Here is a warning"}} alignValidationMessage="right" />
```
