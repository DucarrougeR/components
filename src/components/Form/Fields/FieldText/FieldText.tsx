import * as React from 'react'
import { FormControlDirections } from '../../FormGroup/FormGroup'
import { InputText, InputTextProps } from '../../Inputs/InputText/InputText'
import { Field } from '../Field'

export class FieldText extends Field<InputTextProps> {
  public render() {
    return super.render()
  }

  protected setLabelAlignment(): FormControlDirections | undefined {
    return this.props.alignLabel
  }

  protected setValidationMethodAlignment(): FormControlDirections | undefined {
    return this.props.alignValidationMessage
      ? this.props.alignValidationMessage
      : 'bottom'
  }

  protected setInput() {
    const {
      label,
      alignLabel,
      validationState,
      alignValidationMessage,
      ...props
    } = this.props

    return (
      <InputText
        validationType={validationState ? validationState.type : undefined}
        {...props}
      />
    )
  }
}
