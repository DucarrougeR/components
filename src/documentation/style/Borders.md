<div class="component-desc"><p>Borders help define the space and structure of the interface. </p></div>

<div class="doc-section-divider"></div>

```js noeditor
const BorderRender = require('../../../styleguide_components/BorderRender').BorderRender;

<BorderRender />
```
