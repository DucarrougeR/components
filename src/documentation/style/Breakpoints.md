<div class="component-desc"><p>Breakpoints give hooks for components to adapt and change in relation to the window size.</p></div>

<div class="doc-section-divider"></div>


### Available Breakpoints

```js noeditor
const BreakpointOptionsRender = require('../../../styleguide_components/BreakpointTable').BreakpointOptionsRender;

<BreakpointOptionsRender />
```


