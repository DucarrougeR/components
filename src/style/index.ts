export * from './animations'
export * from './helpers'
export * from './theme'
export * from './styled_components'
