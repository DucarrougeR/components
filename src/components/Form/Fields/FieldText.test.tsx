import { mount } from 'enzyme'
import * as React from 'react'
import { createWithTheme } from '../../../../test/utils/create_with_theme'
import { theme, ThemeProvider } from '../../../style'
import { FieldText } from './FieldText'

test('A FieldText', () => {
  const component = createWithTheme(
    <FieldText label="👍" name="thumbsUp" id="thumbs-up" />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Should accept a value', () => {
  const wrapper = mount(
    <ThemeProvider theme={theme}>
      <FieldText
        label="👍"
        name="thumbsUp"
        id="thumbs-up"
        value="foobar"
        readOnly
      />
    </ThemeProvider>
  )

  const input = wrapper.find('input')
  expect(input.prop('value')).toEqual('foobar')
})

test('Should trigger onChange handler', () => {
  let counter = 0
  const handleChange = () => counter++

  const wrapper = mount(
    <ThemeProvider theme={theme}>
      <FieldText
        label="👍"
        name="thumbsUp"
        id="thumbs-up"
        value="foobar"
        onChange={handleChange}
      />
    </ThemeProvider>
  )

  wrapper.find('input').simulate('change', { target: { value: '' } })
  expect(counter).toEqual(1)
})
