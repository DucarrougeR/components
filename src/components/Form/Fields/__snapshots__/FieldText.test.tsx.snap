// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A FieldText 1`] = `
.c2 {
  border: solid 1px #C1C6CC;
  height: 28px;
  padding: 0 0.5rem;
  border-radius: 4px;
  font-size: 0.875rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 0.5rem;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  border: 0;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  font-weight: 600;
  color: #262D33;
  margin-right: 0.5rem;
  font-size: 0.875rem;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor="thumbs-up"
  >
    üëç
  </label>
  <input
    className="c2"
    id="thumbs-up"
    name="thumbsUp"
    type="text"
  />
</div>
`;
